website/client-old/js/services/groupServices.js

```diff
@@ -107,8 +107,12 @@ angular.module('habitrpg')
       });
     };
 
-    $rootScope.$on('syncPartyRequest', function () {
-      party(true);
+    $rootScope.$on('syncPartyRequest', function (event, options) {
+      if (options.type === 'user_update') {
+        var index = _.findIndex(data.party.members, function(user) { return user._id === options.user._id; });
+        var memberToUpdate = data.party.members[index];
+        _.assign(memberToUpdate, User.user);
```

Comment:
Can we change this so that it only syncs the fields that are fetched from the server in first place? The members array only contains a few fields, not the entire user object which is very big (contains tasks too on client) and will slow down Angular's change detection